### ISSUE TITLE
Incorrect too-many-function-args for a function defined and called within a class

### ISSUE THREAD

**Issue by nedbat** [PyCQA/pylint#6592]
### Bug description

```python
class Dyn:
    def _make_method(name):     # pylint: disable=no-self-argument
        def _the_method(self):
            print(name)
        return _the_method

    print_nedbat = _make_method("nedbat")

Dyn().print_nedbat()
```
This code works. But pylint doesn't like it:
```
% pylint -E weirdmethod.py
************* Module weirdmethod
weirdmethod.py:7:19: E1121: Too many positional arguments for method call (too-many-function-args)
```

### Command used

```shell
pylint -E weirdmethod.py
```

### Pylint output

```shell
************* Module weirdmethod
weirdmethod.py:7:19: E1121: Too many positional arguments for method call (too-many-function-args)
```

### Expected behavior

Since the code works, this is not an error. Pylint shouldn't complain.

### Pylint version

```shell
pylint 2.13.8
astroid 2.11.5
Python 3.7.13 (default, Mar 16 2022, 20:45:04)
[Clang 12.0.0 (clang-1200.0.32.29)]

(btw, this didn't happen with pylint 2.13.7 and astroid 2.11.3)
```

### OS / Environment

Mac OS 10.15.7

**Comment by jacobtylerwalls** [comment]
Thanks for the report. The behavior changed in PyCQA/astroid@1531. I think the new result is closer to what pylint does in general, which is to complain when `@staticmethod` decorators are missing. When I add `@staticmethod`, the message goes away. I take the point, though, that the decorator is not needed at runtime!

At the very least emitting a lower severity message for this scenario is something to consider.

**Comment by nedbat** [comment]
Huh, on 3.7-3.9, a staticmethod produces "TypeError: 'staticmethod' object is not callable".  On 3.10 it works...

**Comment by da-dada** [comment]
@nedbat, for < 3.10
````python
class Dyn:
    @staticmethod
    def _make_method(name):
        def _the_method(self):
            print(name)
        return _the_method

print_nedbat = Dyn()._make_method("nedbat")
print_nedbat(print_nedbat)
````
